### ISSUE TITLE
Issue with docking procedure

### ISSUE THREAD

**Issue by GrignardReagent** [asapdiscovery/asapdiscovery#1111]
I've run into issues with the docking procedure, using codes from [here](https://github.com/asapdiscovery/asapdiscovery/blob/main/examples/docking_and_scoring.ipynb).
I was docking a new ligand into the Mpro protein, and I encountered problems with the  `results = docker.dock(inputs=[input_pair])` part of my code:
![image](https://github.com/asapdiscovery/asapdiscovery/assets/69822082/debdcd02-e5a0-44f6-abeb-507f783b97b3)

Not sure why I couldnt upload a pdb file but it is the Mpro-x0072 protein-ligand complex from COVID moonshot that I used. 
```
# This is how I prepped my protein: 
!asap-cli protein-prep --target SARS-CoV-2-Mpro --pdb-file ./Mpro-x0072_0.pdb --output-dir . --no-save-to-cache

from asapdiscovery.data.testing.test_resources import fetch_test_file
from asapdiscovery.data.schema.complex import Complex, PreppedComplex
from asapdiscovery.data.schema.ligand import Ligand

# grab a pre-prepared complex 
prepped_complex = PreppedComplex.from_oedu_file("./Mpro-x0072_0-203b062aff4311be394b0b33fa21b555c912f269401b4f726ab77be8621a05ec+JGDDFCYMSLNOGJ-UHFFFAOYNA-N/Mpro-x0072_0.oedu", 
                    target_kwargs={"target_name": "SARS-CoV-2-Mpro", "target_hash": "mock_hash"})

from rdkit import Chem
from rdkit.Chem import AllChem
molecule = Chem.MolFromSmiles("[H]N=c1c(C([H])([H])C([H])([H])N([H])C(=O)c2c([H])c([H])c([H])c([H])c2F)c(O[H])c2c([H])c([H])c([H])c(N([H])[H])c2n1N([H])C(=O)N([H])c1c([H])c([H])c([H])c([H])c1[H]")

# Function to remove stereochemistry
def remove_stereochemistry(mol):
    Chem.RemoveStereochemistry(mol)
    return mol

# Remove stereochemistry from the molecule
molecule_no_stereo = remove_stereochemistry(Chem.Mol(molecule))

# Convert back to SMILES to verify
no_stereo_smiles = Chem.MolToSmiles(molecule_no_stereo)

ligand = Ligand.from_smiles(no_stereo_smiles, compound_name="molecules_bx_2024_06_18_154410") # compound name is mandatory

# lets go ahead and make a Docker object
docker = POSITDocker()

from asapdiscovery.docking.openeye import POSITDocker
from asapdiscovery.docking.docking import DockingInputPair
input_pair = DockingInputPair(ligand=ligand, complex=prepped_complex)

results = docker.dock(inputs=[input_pair])
```
I've tried removing the stereochemistry using rdkit and it's still the same... Same when I used OpenEye to do this:
```
from openeye.oechem import *
def remove_stereochemistry(oe_mol):
    for atom in oe_mol.GetAtoms():
        if atom.IsChiral():
            atom.SetStereo(OEAtomStereo_Tetrahedral, OEAtomStereo_Undefined)
    for bond in oe_mol.GetBonds():
        if bond.HasStereoSpecified():
            bond.SetStereo(OEBondStereo_CisTrans, OEBondStereo_Undefined)

# Example SMILES string
molecule = "[H]N=c1c(C([H])([H])C([H])([H])N([H])C(=O)c2c([H])c([H])c([H])c([H])c2F)c(O[H])c2c([H])c([H])c([H])c(N([H])[H])c2n1N([H])C(=O)N([H])c1c([H])c([H])c([H])c([H])c1[H]"

# Convert SMILES string to OEChem molecule
oe_mol = OEMol()
OESmilesToMol(oe_mol, molecule)
remove_stereochemistry(oe_mol)
no_stereo_smiles = OECreateSmiString(oe_mol)
```

**Comment by hmacdope** [comment]
Hey @GrignardReagent  and welcome! Happy to help (also happy to jump on a call if you need). 


Just a quick clarification on terminology as I think this may be causing some misunderstanding, leading you to try and do the reverse operation to the one you want. 

There is a difference between "removing" and "fully specifying" stereochemistry. In fact the two are approximately the opposite operations of each other.  The error from `omega` telling you not that you need to **fully specify** the chemistry on your input molecule, not **remove** stereochemistry information. For example compare the following smiles strings:

S-Alanine: `N[C@@H](C)C(=O)O`
R-Alanine: `N[C@H](C)C(=O)O`

These have **defined stereochemistry**, meaning that their 3D structure can roughly be inferred  from their SMILES entry as the ordering of the substituents is defined (remember the Cahn-Ingold-Prelog (CIP) [rules](https://en.wikipedia.org/wiki/Cahn%E2%80%93Ingold%E2%80%93Prelog_priority_rules)

However the following doesn't have defined stereochemistry, meaning it could be *R* or *S*

Alanine: NC(C)C(=O)O

Now think of this from the perspective of the conformer generation algorithm in `omega`, followed by docking. Which isomer is it trying to generate conformers for? To avoid confusing the user and mixing conformers where not intended omega fails where the stereochemistry is not defined. You can turn this off (more on this below). 

To summarise this error is telling you to add **more** stereochemistry information to your input molecules rather than take stereochemistry information away. 

**HOWEVER**: 

A recent PR changed things to allow non-defined stereochemistry to not fail at the omega step (#1108) which should fix this   would you be able to try with the most recent `main`? You will need to clone the repo and do a developer install. Give it a try or I can help you if unclear.

**Comment by hmacdope** [comment]
@GrignardReagent did this help?

**Comment by hmacdope** [comment]
Closing as stale
